pkg/nodediscovery/nodediscovery.go

```diff
@@ -59,6 +59,12 @@ type k8sNodeGetter interface {
 	GetK8sNode(ctx context.Context, nodeName string) (*corev1.Node, error)
 }
 
+// The KVStoreNodeUpdater interface is used to provide an abstraction for the
+// NodeStore object logic used to update a node entry in the KV store.
+type KVStoreNodeUpdater interface {
+	UpdateKVNodeEntry(node *nodeTypes.Node) error
+}
+
```

Comment:
Does this interface need to exist if it is already defined in `pkg/k8s/watchers/node.go`? I don't see it used in the rest of the diff for nodediscovery?

Reply:
Right, no, this can be dropped (I have another branch which modifies this file, I'll do it there)